codejail/safe_exec.py

```diff
@@ -256,16 +205,6 @@ def filter_unserializable(obj):
                 except Exception:
```

Comment:
It took me a long time to understand why this pull request fixes the problem.  The one bit of code that wasn't changed was exactly the code that seemed to be involved (dicts).  filter_unserializable is a strange function: it's recursive, and returns cleaned data, but sometimes raises an exception (in the case of decoding non-utf8).  It's that exception we're counting on.  It's counter-intuitive that it removes some bad data, but throws an exception on other bad data.  Can we at least explain it in a docstring?

Also, how it removes data is surprising.  The previous way made some sense, even though it wasn't good for other reasons.  This needs explaining.

Reply:
Or, it needs more edits.  I hope it's OK that this is pushing us back toward the original code, but what do you think of this:
- Change json_safe to explicitly be for dicts only (the argument is called d already): json_safe_dict.
- decode_object will be a recursive function that either fully converts its object to utf8-decode any bytestrings it finds, or it raises an exception.
- json_safe_dict would call decode_object to fix its keys and values. If decode_object raised an exception, it would skip that key/value pair.

To me, this makes the code easier to explain and understand.